### DISCUSSION TITLE
Whats the recommended way to handle/decode errors from cross contract calls?

### DISCUSSION THREAD

**Discussion by oveddan** [wevm/viem#1198] [General]
Given you have a contract that calls another contract, sometimes the other contract reverts with a custom error.  However, the calling contract does not have that error in it's abi, so viem cannot decode that error result.

Example:

```solidity
contract CalleeContract {
    error Oops();

    function callMe() external {
        revert Oops();
    }
}

contract CallingContract {
    CalleeContract calleeContract;

    constructor() {
        calleeContract = new CalleeContract();
    }

    function callCalle() external {
        calleeContract.callMe();
    }
}
```

What's the recommend approach to be able to decode errors from the `CalleeContract` when calling it from the `CallerContract` using `viem`?

**Comment by dsimmons** [comment]
Also seeing this, would love to know the best way to handle it! :slightly_smiling_face:

**Comment by junkim012** [comment]
Curious if you found a viem native solution for this problem? 

The two things I do right now is either 1. move all custom errors to a single contract and inherit it such that all of the possible errors show up on the calling contracts' ABI. 2. Iterate through all possible ABI's with viem's `decodeErrorResult` until it finds the valid ABI with the custom error that was thrown.

**Comment by maguroid** [comment]
+1
